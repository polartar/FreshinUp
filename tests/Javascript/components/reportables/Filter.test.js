import { createLocalVue, shallowMount, mount } from '@vue/test-utils'
import Component from '~/components/reportables/Filter.vue'

describe('Modifier', () => {
  // Component instance "under test"
  let localVue
  describe('Snapshots', () => {
    test('defaults', () => {
      localVue = createLocalVue()
      const wrapper = mount(Component, {
        localVue: localVue
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })
  describe('Methods', () => {
    beforeEach(() => {
      localVue = createLocalVue()
    })
    test('run() emits runFilter', () => {
      const wrapper = shallowMount(Component)
      wrapper.vm.run({ term: 'test' })
      expect(wrapper.emitted().runFilter).toBeTruthy()
      expect(wrapper.emitted().runFilter).toHaveLength(1)
      expect(wrapper.emitted().runFilter[0]).toEqual([{
        name: 'test'
      }])
    })
  })
})
