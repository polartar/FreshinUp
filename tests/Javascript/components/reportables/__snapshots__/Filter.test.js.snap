// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Snapshots defaults 1`] = `
<div
  autocomplete_url="/api/foodfleet/financial-reports"
  class="v-card v-sheet theme--light white"
>
  <form
    class="v-form"
    novalidate="novalidate"
  >
    <div
      class="container filter-sorter-container fluid"
    >
       
      <div
        class="layout filter-sorter-main-layout row wrap"
      >
        <!---->
         
        <div
          class="flex grow"
        >
          <div
            class="v-input v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--placeholder theme--light"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <input
                    placeholder="Search"
                    type="text"
                  />
                </div>
                <div
                  class="v-input__append-inner"
                >
                  <div
                    class="v-input__icon v-input__icon--append"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon material-icons theme--light"
                    >
                      search
                    </i>
                  </div>
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <!---->
         
        <!---->
         
      </div>
       
      <!---->
       
      <div
        class="layout filter-sorter-expanded-layout align-center justify-center row fill-height"
        style="display: none;"
      >
        <div
          class="flex"
        />
         
        <div
          class="flex text-no-wrap"
        >
          <button
            class="v-btn v-btn--large theme--light"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
            Clear Filters
          
            </div>
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;
